### PR TITLE
Fix #727 Error message from SubQuerySQLSyntaxProvider#apply(name: SQLSyntax) is confusing

### LINKED ISSUE #727: Error message from SubQuerySQLSyntaxProvider#apply(name: SQLSyntax) is confusing
I guess `SubQuerySQLSyntaxProvider#apply(name: SQLSyntax)` expects a result name SQLSyntax as a parameter.

This method throws an error for a column name (not result name) and the error message seems inappropriate. It shows the `name` is wrong, but `name` is in the `registered names`.

```scala
import org.slf4j.LoggerFactory
import scalikejdbc._


case class Account(id: Int, name: String)
object Account extends SQLSyntaxSupport[Account]


object Hello {

  private val logger = LoggerFactory.getLogger(getClass)

  def main(args: Array[String]): Unit = {
    Class.forName("org.h2.Driver")
    ConnectionPool.singleton("jdbc:h2:mem:hello", "user", "pass")

    DB.autoCommit { implicit session =>
      sql"create table account(id integer, name varchar(10))".execute.apply()
    }

    val a = Account.syntax("a")
    val s = SubQuery.syntax("s", a.resultName)

    println(s(a.resultName.id)) // => SQLSyntax(value: s.i_on_a, parameters: List())

    // scalikejdbc.InvalidColumnNameException: Invalid column name. (name: id, registered names: id,name)
    println(s(sqls"id")) 
  }

}
```

### PR DESCRIPTION
**Pull request by seratch**
This pull request improves the error message when giving an invalid value to sub query syntax object.

### CODE REVIEW

**Comment by xuwei-k** [comment]
```
[info] - should convert timeZone *** FAILED ***
[info]   org.h2.jdbc.JdbcSQLException: Function "DATE_FORMAT" not found; SQL statement:
[info] select date_format(time, '%Y-%m-%d %H:%i:%S') as s from qi_time_holder where id = 1 [90022-196]
[info]   at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
[info]   at org.h2.message.DbException.get(DbException.java:179)
[info]   at org.h2.message.DbException.get(DbException.java:155)
[info]   at org.h2.command.Parser.readJavaFunction(Parser.java:2487)
[info]   at org.h2.command.Parser.readFunction(Parser.java:2539)
[info]   at org.h2.command.Parser.readTerm(Parser.java:2876)
[info]   at org.h2.command.Parser.readFactor(Parser.java:2390)
[info]   at org.h2.command.Parser.readSum(Parser.java:2377)
[info]   at org.h2.command.Parser.readConcat(Parser.java:2347)
[info]   at org.h2.command.Parser.readCondition(Parser.java:2178)
[info]   ...
[info] - should have result.name
[info] Run completed in 4 seconds, 259 milliseconds.
[info] Total number of tests run: 27
[info] Suites: completed 7, aborted 0
[info] Tests: succeeded 26, failed 1, canceled 0, ignored 0, pending 0
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error] 	scalikejdbc.QueryInterfaceSpec
[error] (interpolation/test:test) sbt.TestsFailedException: Tests unsuccessful
```

https://travis-ci.org/scalikejdbc/scalikejdbc/jobs/258216610#L3423

**Comment by tototoshi** [comment]
SubQueryResultSQLSyntaxProvider and SubQueryResultNameSQLSyntaxProvider also have this problem.

**Comment by seratch** [comment]
Weird. The build on my side looks successful. https://travis-ci.org/seratch/scalikejdbc/builds/276170503

I'd like to try to create another pull request.

File: scalikejdbc-interpolation/src/test/scala/scalikejdbc/SubQuerySpec.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package scalikejdbc
+
+import org.scalatest._
+
+class SubQuerySpec extends FlatSpec with Matchers with SQLInterpolation {
+
+  Class.forName("org.h2.Driver")
+  ConnectionPool.singleton("jdbc:h2:mem:SubQuerySpec", "user", "pass")
```
Comment (seratch) [inline scalikejdbc-interpolation/src/test/scala/scalikejdbc/SubQuerySpec.scala]: Ah, I figured out the reason why builds fail. This line breaks other tests.

**Pull request merged**